### PR TITLE
schedule should be a dict in opts

### LINKED ISSUE #44378: minion: infinite loop during start when schedule key is null
### Description of Issue/Question
In an minion's config file, if the schedule key is null, the salt-minion will never successfully start.
add_job() assumes the value is iterable, throws an exception and the caller calls add_job() again.

### Steps to Reproduce Issue
echo 'schedule:' >> /etc/salt/minion
*restart minion*
view the minion logs
```
salt-minion[18002]: [CRITICAL] Unexpected error while connecting to salt
salt-minion[18002]: Traceback (most recent call last):
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/salt/minion.py", line 876, in _connect_minion
salt-minion[18002]:     yield minion.connect_master(failed=failed)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
salt-minion[18002]:     value = future.result()
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
salt-minion[18002]:     raise_exc_info(self._exc_info)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
salt-minion[18002]:     yielded = self.gen.throw(*exc_info)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/salt/minion.py", line 1054, in connect_master
salt-minion[18002]:     yield self._post_master_init(master)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
salt-minion[18002]:     value = future.result()
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
salt-minion[18002]:     raise_exc_info(self._exc_info)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/tornado/gen.py", line 1069, in run
salt-minion[18002]:     yielded = self.gen.send(value)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/salt/minion.py", line 1109, in _post_master_init
salt-minion[18002]:     }, persist=True)
salt-minion[18002]:   File "/usr/lib/python2.7/site-packages/salt/utils/schedule.py", line 566, in add_job
salt-minion[18002]:     elif new_job in self.opts['schedule']:
salt-minion[18002]: TypeError: argument of type 'NoneType' is not iterable
```

### Versions Report
Salt Version:
           Salt: 2016.11.6
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.26.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, Sep 20 2017, 01:25:59)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.2.2
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.13.8-1-ARCH
         system: Linux
        version: Not Installed

### PR DESCRIPTION
**Pull request by gtmanfred**
### What issues does this PR fix or reference?
Fixes #44378

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by rallytime** [comment]
@gtmanfred There are some related test failures on this change.

**Review by garethgreenaway — APPROVED** [review]

**Comment by cachedout** [comment]
re-run py2

**Comment by cachedout** [comment]
This is interesting. It seems like those test failures might be legitimate. @gtmanfred can you investigate?

**Comment by gtmanfred** [comment]
Looks like it was possible that the schedule would not be set.

I have added schedule to the VALID_OPTS in salt.config so that it will warn when it is set incorrectly, and also set it to a default of {}

**Comment by garethgreenaway** [comment]
re-run lint

**Comment by gtmanfred** [comment]
re-run lint

File: salt/config/__init__.py [hunk 1]
```diff
@@ -1223,6 +1226,7 @@ def _gather_buffer_space():
     'cache': 'localfs',
     'salt_cp_chunk_size': 65536,
     'minion_sign_messages': False,
+    'schedule', {}
```
Comment (rallytime) [inline salt/config/__init__.py]: This should be `'schedule': {},` instead of a `,`.

**Pull request merged**